docs/improving-gci-experience.md

```diff
@@ -0,0 +1,350 @@
+# Introduction
+
+Whether you are a new or experienced GCI participant, a GCI mentor or Zulip
+maintainer or a Zulip contributor who wants to know a bit more about our GCI
```

Comment:
I think I was trying to do it in pairs, but I've had an uneven number of ideas. Will rework this, thanks!